### PR TITLE
34 datatakeid parameter not included in the mph of rawspid pc products

### PR DESCRIPTION
**Pull request by mivermeu**
This PR addresses a change in the specifications concerning RAWS products. These are now required to have the `bio:dataTakeID` parameter set in their MPH (issue #34). Data take information getting is generalised in preparation for support in generators other than L0 and RAWS.

### CODE REVIEW

File: procsim/biomass/test/test_raw_product_generator.py [hunk 1]
```diff
@@ -224,6 +224,84 @@ def testMergeToSingleSlice(self):
...
+        end = datetime.datetime(2021, 2, 1, 0, 29, 32, 0, tzinfo=datetime.timezone.utc)
+        gen._hdr.validity_start = gen._hdr.begin_position = begin
+        gen._hdr.validity_stop = gen._hdr.end_position = end
+
+        with self.assertRaises(ScenarioError):
+            gen.generate_output()
+
+    def testMultipleDataTakes(self) -> None:
+        '''
+        Test that multiple data takes appear in the MPH and that the
```
Comment (pe1obw) [inline procsim/biomass/test/test_raw_product_generator.py:245]: In the current implementation, only a single datatake_id can appear in a MPH, right?

**Review by pe1obw — APPROVED** [review]
Ok for me!

File: procsim/biomass/product_generator.py [hunk 2]
```diff
@@ -305,6 +307,53 @@ def _get_slice_frame_interval(self,
...
+        sensing_stop = self._hdr.end_position
+        if sensing_start is None or sensing_stop is None:
+            raise ScenarioError('Sensing start and stop time are not set.')
+
+        data_takes = self._scenario_config.get('data_takes')
+        if not data_takes:
+            # No explicit data takes found, use general config.
+            self._logger.info('No data takes found, using general config.')
+            data_takes = [{**self._scenario_config, 'start': self._time_as_iso(sensing_start), 'stop': self._time_as_iso(sensing_stop)}]
+        # Overwrite general config with data take config.
```
[the remark this answers is not in the document]
Comment (mivermeu) [reply procsim/biomass/product_generator.py:330]: Agreed. I've altered the comment to be more accurate and used an `if else` here, although of course it shouldn't matter in practice.

File: procsim/biomass/test/test_raw_product_generator.py (hunk 1 above)
Comment (mivermeu) [reply procsim/biomass/test/test_raw_product_generator.py]: Indeed, this was an outdated docstring! I've updated it.

**Pull request merged**